src/roiextractors/extractors/tiffimagingextractors/scanimagetiffimagingextractor.py

```diff
@@ -22,13 +24,487 @@ def _get_scanimage_reader() -> type:
+
+    Returns
+    -------
+    metadata_parsed: dict
+        Dictionary of parsed metadata.
+
+    Notes
+    -----
+    SI.hChannels.channelsActive = '[1;2;...;N]' where N is the number of active channels.
+    SI.hChannels.channelName = "{'channel_name_1' 'channel_name_2' ... 'channel_name_M'}"
+        where M is the number of channels (active or not).
+    """
+    sampling_frequency = float(metadata["SI.hRoiManager.scanVolumeRate"])
+    num_channels = len(metadata["SI.hChannels.channelsActive"].split(";"))
+    num_planes = int(metadata["SI.hStackManager.numSlices"])
+    frames_per_slice = int(metadata["SI.hStackManager.framesPerSlice"])
+    channel_names = metadata["SI.hChannels.channelName"].split("'")[1::2][:num_channels]
+    metadata_parsed = dict(
+        sampling_frequency=sampling_frequency,
+        num_channels=num_channels,
+        num_planes=num_planes,
+        frames_per_slice=frames_per_slice,
+        channel_names=channel_names,
+    )
+    return metadata_parsed
+
+
+def parse_metadata_v3_8(metadata):
+    """Parse metadata dictionary to extract relevant information and store it standard keys for ImagingExtractors.
+
+    Requires old version of metadata (v3.8).
+    Currently supports
+    - sampling frequency
+    - num_channels
+    - num_planes
+
+    Parameters
+    ----------
+    metadata : dict
+        Dictionary of metadata extracted from the TIFF file.
+
+    Returns
+    -------
+    metadata_parsed: dict
+        Dictionary of parsed metadata.
+    """
+    sampling_frequency = float(metadata["state.acq.frameRate"])
+    num_channels = int(metadata["state.acq.numberOfChannelsSave"])
+    num_planes = int(metadata["state.acq.numberOfZSlices"])
+    metadata_parsed = dict(
+        sampling_frequency=sampling_frequency,
+        num_channels=num_channels,
+        num_planes=num_planes,
+    )
+    return metadata_parsed
+
+
+class MultiPlaneImagingExtractor(ImagingExtractor):
+    """Class to combine multiple ImagingExtractor objects by depth plane."""
+
+    extractor_name = "MultiPlaneImaging"
+    installed = True
+    installatiuon_mesage = ""
+
+    def __init__(self, imaging_extractors: List[ImagingExtractor]):
+        """Initialize a MultiPlaneImagingExtractor object from a list of ImagingExtractors.
+
+        Parameters
+        ----------
+        imaging_extractors: list of ImagingExtractor
+            list of imaging extractor objects
+        """
+        super().__init__()
+        assert isinstance(imaging_extractors, list), "Enter a list of ImagingExtractor objects as argument"
+        assert all(isinstance(imaging_extractor, ImagingExtractor) for imaging_extractor in imaging_extractors)
+        self._check_consistency_between_imaging_extractors(imaging_extractors)
+        self._imaging_extractors = imaging_extractors
+        self._num_planes = len(imaging_extractors)
+
+    def _check_consistency_between_imaging_extractors(self, imaging_extractors: List[ImagingExtractor]):
```

Comment:
Just something to be aware of, from prior experience...

While this check is very valuable from a perspective of safety of operation support on our end, for a very large number of relatively small files (may or may not be the case depending on the buffering size here) this check can be an extremely inefficient slowdown since it has to open/read a bit of info for every single I/O

In some previous conversions I built in an override via an boolean argument to disable the check for speedup, basically saying 'I trust the data to be consistent'

Reply:
I don't understand why this would be slow.  All of the getters just return pre-stored attributes ex. 
```
def get_sampling_frequency(self) -> float:
        return self._sampling_frequency
```
So there shouldn't be any I/O during the consistency check -- all of it should happen during the construction of `SinglePlaneImagingExtractors`.  Maybe the other IE's you were working with were designed differently?

Reply:
Sorry for not being specific ; the skip I built in was to the `__init__` of the extractors so that they did not fetch/store the attribute to being with. But they were mostly doing that already in order to pass the consistency check. So the skip override does both, (i) does not fetch the attributes at time of `__init__` (only on call to retrieve each specific attribute) and (ii) skip the consistency check that would have run on those

Reply:
I feel like this is an architectural change/option that we would want to propagate to all the IE's, maybe accompanied by changing the private attributes to true properties.

Reply:
Riffing a bit: We could add an option to the base `ImagingExtractor` class for `precomputed_properties=True` and then if you pre-compute the properties, I/O happens during `__init__` and they get stored as attributes.  But if you choose not to (like in the case you're suggesting) then I/O happens lazily and you can skip the consistency checks for MultiImagingExtractors.

Reply:
Regardless, I think it's outside of the scope of this PR.

Reply:
> I see bc if you have a multi-file setup, you only need the first IE to actually load the properties, so all of the other I/O calls are just for checking consistency -- right?

Exactly - which is a good thing in general when data volumes are small enough to not matter much for performance at runtime. There's probably a threshold of number of files vs. size of each file + time taken to parse metadata in each


> I feel like this is an architectural change/option that we would want to propagate to all the IE's, maybe accompanied by changing the private attributes to true properties.
> Regardless, I think it's outside of the scope of this PR.

Yes, I just wanted you to be aware of it in general

Reply:
See #247